### PR TITLE
PHP performance improvements

### PR DESCRIPTION
**Pull request by ctrl-f5**
- `isset()` is up to 20% faster than `array_key_exists()`
- `preg_split` can ignore empty strings
- removed unnecessary string conversion 
- added output buffering

on a smaller dataset, I got more than 30% increased performance with these changes.

### CODE REVIEW

**Comment by juditacs** [comment]
test3 and test4 fail.

File: php/wordcount.php [hunk 1]
```diff
@@ -7,37 +7,30 @@
...
-
-
 }
 fclose($stdin);
 $array2 = array();
-
 foreach($array as $key => $value){
-
-    if (!array_key_exists($value,$array2)){
+    if (!isset($array2[$value])){
```
Comment (ircmaxell) [inline php/wordcount.php]: This if statement is redundant. This entire loop can be simplified to:

    foreach ($array as $key => $value) {
        $array2[$value][] = $key;
    }

**Comment by GrahamCampbell** [comment]
> isset() is up to 20% faster than array_key_exists()

Only on php 5.x. Not on 7.x.

**Comment by GrahamCampbell** [comment]
Also, what happens if we run these benchmarks on php 7.x?

**Comment by GrahamCampbell** [comment]
PHP has some internal changes to "array" storage that should make a difference.

**Comment by tpunt** [comment]
> Only on php 5.x. Not on 7.x.

I'm not sure what the performance difference is now, but `isset()` is still more performant than `array_key_exists()`.

> Also, what happens if we run these benchmarks on php 7.x?

These benchmarks are run on PHP 7.

**Pull request closed without merging**